lib/Mojo/Util.pm

```diff
@@ -37,11 +38,23 @@ use constant {
 # To generate a new HTML entity table run this command
 # perl examples/entities.pl
 my %ENTITIES;
-for my $line (split "\n", join('', <DATA>)) {
-  next unless $line =~ /^(\S+)\s+U\+(\S+)(?:\s+U\+(\S+))?/;
-  $ENTITIES{$1} = defined $3 ? (chr(hex $2) . chr(hex $3)) : chr(hex $2);
+{
+  # Don't use Mojo::File here due to circular dependencies
+  my $html_path
+    = File::Spec->catfile(File::Basename::dirname(Cwd::realpath(__FILE__)),
+    'resources', 'html_entities.txt');
+
+  local $/ = undef;
```

Comment:
Why do this on an extra line instead of using the `do` idiom?

Reply:
Like this?
`my $file_content = do { local $/; <$fh> };`

I wasn't familiar with that pattern, I had to Google it.

Reply:
Yes, that's the standard way of using `local $/` actually.